### PR TITLE
Fill dataset structure from crawler

### PR DESCRIPTION
**Pull request by elic-eon**
### 🤔 Why?

Fill out the three level `DatasetStructure` to enable the next step of showing case sensitive name. Potentially we can support name with dots.

### 🤓 What?

- Fill out the field for `Bigquery`, `Snowflake`, `Redshift`, `PostgreSQL`, and `Glue`.
- Use `dataset_fullname` to fill out the logical id as possible

### 🧪 Tested?

- Unit tests.
- Run `Snowflake`, `Bigquery`, and `Redshift` crawler on local, ingest the MCEs, and verified the result on Metaphor.

### CODE REVIEW

File: metaphor/bigquery/profile/extractor.py [hunk 1]
```diff
@@ -139,7 +140,9 @@ def job_callback(job: QueryJob, schema: DatasetSchema, dataset: Dataset):
             jobs.remove(job.job_id)
 
         dataset = BigQueryProfileExtractor._init_dataset(
-            f"{table.project}.{table.dataset_id}.{table.table_id}"
+            dataset_fullname(
```
Comment (mars-lan) [inline metaphor/bigquery/profile/extractor.py:143]: Let's rename this to `dataset_normalized_name` for clarify?

File: metaphor/glue/extractor.py [hunk 2]
```diff
@@ -139,12 +141,14 @@ def _get_tables(self, database: str):
...
         self,
-        full_name: str,
+        schema: str,
+        table_name: str,
         table_type: str,
         description: str,
         row_count: int,
         last_updated: datetime,
     ) -> Dataset:
+        full_name = f"{schema}.{table_name}"
```
Comment (mars-lan) [inline metaphor/glue/extractor.py:151]: ditto

**Review by mars-lan — APPROVED** [review]
Otherwise LGTM. Thanks for fixing this.

File: metaphor/bigquery/profile/extractor.py (hunk 1 above)
Comment (elic-eon) [reply metaphor/bigquery/profile/extractor.py]: SG. Updated.

File: metaphor/glue/extractor.py (hunk 2 above)
Comment (elic-eon) [reply metaphor/glue/extractor.py]: Fixed.

File: metaphor/postgresql/profile/extractor.py [hunk 3]
```diff
@@ -203,14 +204,19 @@ async def _create_connection_pool(self) -> asyncpg.Pool:
...
         description: str,
         row_count: int,
         table_size: int,
     ) -> None:
         """Overwrite PostgreSQLExtractor._init_dataset"""
-        super()._init_dataset(full_name, table_type, description, row_count, table_size)
+        super()._init_dataset(
+            database, schema, table, table_type, description, row_count, table_size
+        )
+        full_name = dataset_normalized_name(database, schema, table)
```
Comment (mars-lan) [inline metaphor/postgresql/profile/extractor.py:219]:
```suggestion
        normalized_name = dataset_normalized_name(database, schema, table)
```

File: metaphor/postgresql/usage/extractor.py [hunk 4]
```diff
@@ -47,7 +47,7 @@ async def extract(self) -> Collection[ENTITY_TYPES]:
                     schema = row["schemaname"]
                     table_name = row["relname"]
                     read_count = row["seq_scan"]
-                    full_name = dataset_fullname(db, schema, table_name)
+                    full_name = dataset_normalized_name(db, schema, table_name)
```
Comment (mars-lan) [inline metaphor/postgresql/usage/extractor.py:50]: ditto

File: metaphor/redshift/extractor.py [hunk 5]
```diff
@@ -68,7 +68,9 @@ async def _fetch_redshift_table_stats(self, conn, catalog: str) -> None:
         )
 
         for result in results:
-            full_name = dataset_fullname(catalog, result["schema"], result["table"])
+            full_name = dataset_normalized_name(
```
Comment (mars-lan) [inline metaphor/redshift/extractor.py:71]: ditto

File: metaphor/snowflake/extractor.py [hunk 6]
```diff
@@ -195,7 +196,7 @@ def _fetch_columns(self, cursor: SnowflakeCursor, database: str) -> None:
             default,
             comment,
         ) in cursor:
-            full_name = dataset_fullname(database, table_schema, table_name)
+            full_name = dataset_normalized_name(database, table_schema, table_name)
```
Comment (mars-lan) [inline metaphor/snowflake/extractor.py:199]: ditto

File: metaphor/snowflake/extractor.py [hunk 7]
```diff
@@ -283,7 +284,7 @@ def _fetch_unique_keys(self, cursor: SnowflakeCursor, database: str) -> None:
                 entry[4],
                 entry[6],
             )
-            table = dataset_fullname(database, schema, table_name)
+            table = dataset_normalized_name(database, schema, table_name)
```
Comment (mars-lan) [inline metaphor/snowflake/extractor.py:287]: ditto

File: metaphor/snowflake/extractor.py [hunk 8]
```diff
@@ -449,12 +450,15 @@ def _build_tag_string(tag_key: str, tag_value: str) -> str:
...
-        full_name: str,
+        database: str,
+        schema: str,
+        table: str,
         table_type: str,
         comment: str,
         row_count: Optional[int],
         table_bytes: Optional[float],
     ) -> Dataset:
+        full_name = dataset_normalized_name(database, schema, table)
```
Comment (mars-lan) [inline metaphor/snowflake/extractor.py:461]: ditto

**Review by mars-lan — COMMENTED** [review]
Please rename all `full_name` to `noramlized_name`

File: metaphor/glue/extractor.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (elic-eon) [reply metaphor/glue/extractor.py]: Fixed

**Review by mars-lan — COMMENTED** [review]
Otherwise LGTM. Please also verify that everything is working as expected after the renaming.

File: metaphor/glue/extractor.py [hunk 9]
```diff
@@ -139,12 +142,14 @@ def _get_tables(self, database: str):
...
         self,
-        full_name: str,
+        schema: str,
+        name: str,
         table_type: str,
         description: str,
         row_count: int,
         last_updated: datetime,
     ) -> Dataset:
+        full_name = dataset_normalized_name(schema=schema, table=name)
```
[the remark this answers is not in the document]
Comment (mars-lan) [reply metaphor/glue/extractor.py]: Argh got it. Missed that part.

**Comment by elic-eon** [comment]
> 

Everything looks find. Verified two version output.

**Pull request merged**